YOUR_ADMIN/zca_bootstrap_colors.php

```diff
@@ -16,23 +16,81 @@
 
 $_GET['gID'] = $groupID->fields['configuration_group_id'];
 
-// BOF download SQL file
-$sqlfile = isset($_GET['sqlfile']) ? $_GET['sqlfile'] : '';
-if (zen_not_null($sqlfile)) {
-  $filename = 'zca_bootstrap_colors_' . date('Ymd_His') . '.sql';
-  header('Content-Type: text/plain; charset=utf-8');
+// BOF upload CSV file
+$uploadcsv = isset($_GET['uploadcsv']) ? $_GET['uploadcsv'] : '';
+if (zen_not_null($uploadcsv)) {
+  $file_contents = '';
+  $color_list = [];
+  if (isset($_FILES['csv_file']) && isset($_FILES['csv_file']['tmp_name']) && $_FILES['csv_file']['tmp_name'] != '') {
+    $filename = $_FILES['csv_file']['tmp_name'];
+    if (($handle = fopen($filename, "r")) !== false) {
+      while (($data = fgetcsv($handle, 1000, ",")) !== false) {
```

Comment:
Before _assuming_ that there are 3 elements provided in the incoming CSV file, I'll suggest a check that the `count($data)` is at least 3 and, if not, bailing with a message to the current admin user.

Reply:
Or at least checking for 2 valid columns. The 3rd isn't important for the import.